allauth/socialaccount/providers/oauth2/client.py

```diff
@@ -30,13 +31,15 @@ def __init__(self, request, consumer_key, consumer_secret,
         self.state = None
         self.headers = None
         self.basic_auth = basic_auth
+        self.api_key = api_key
 
     def get_redirect_url(self, authorization_url, extra_params):
         params = {
             'client_id': self.consumer_key,
             'redirect_uri': self.callback_url,
             'scope': self.scope,
-            'response_type': 'code'
+            'response_type': 'code',
+            'api_key': self.api_key
```

Comment:
It is already possible to pass along additional parameters via the `extra_params`, I would prefer that above adding survey monkey specific keys in here.